src/odemis/acq/test/flim_test.py

```diff
@@ -0,0 +1,218 @@
+            time.sleep(0.3)
+
+    def test_repetitions_and_roi(self):
+        logging.debug("Testing repetitions and roi")
+
+        helper = stream.ScannedTCSettingsStream('Stream', self.detector, self.ex_light, self.lscanner,
+                                                self.sft, self.apd, self.tc_scanner)
+
+        helper.dwellTime.value = 1e-3  # seconds
+        helper.repetition.value = (64, 64)
+        helper.roi.value = (0, 0, 0.1, 0.1)
+        self.assertEqual(helper.scanner.resolution.value, (64, 64))
+        helper.repetition.value = (256, 64)
+        helper.roi.value = (0.15, 0.15, 0.8, 0.8)
+        self.assertEqual(helper.scanner.resolution.value, (256, 64))
+
+    def test_cancelling(self):
+        logging.debug("Testing cancellation")
+
+        helper = stream.ScannedTCSettingsStream('Stream', self.detector, self.ex_light, self.lscanner,
+                                                self.sft, self.apd, self.tc_scanner)
+
+        remote = stream.ScannedRemoteTCStream("remote", helper)
+
+        # Configure values and test acquisition for several dwell times.
+        for dwellTime in (10e-3, 100e-3, 1):
+            helper.dwellTime.value = dwellTime  # seconds
+            helper.repetition.value = (64, 64)
+            helper.roi.value = (0.15, 0.6, 0.8, 0.8)
+            f = remote.acquire()
+            time.sleep(0.5)
+            f.cancel()
+            self.assertTrue(f.cancelled())
+            time.sleep(0.3)
+
+    def test_setting_stream(self):
+
+        helper = stream.ScannedTCSettingsStream('Stream', self.detector, self.ex_light, self.lscanner,
+                                                self.sft, self.apd, self.tc_scanner)
+        spots = stream.SpotScannerStream("spot", helper.tc_detector,
+                                     helper.tc_detector.data, helper.scanner)
+        # Test start and stop of the apd.
+        self._image = None
+        helper.image.subscribe(self._on_image)
+
+        spots.should_update.value = True
+        spots.is_active.value = True
+
+        # shouldn't affect
+        helper.roi.value = (0.15, 0.6, 0.8, 0.8)
+        helper.repetition.value = (1, 1)
+
+        helper.dwellTime.value = 0.1  # s
+        helper.windowPeriod.value = 10  # s
+
+        # Start acquisition
+        helper.should_update.value = True
+        helper.is_active.value = True
+
+        # move spot
+        time.sleep(8.0)
+        spots.roi.value = (0.1, 0.3, 0.1, 0.3)
+        time.sleep(2.0)
+        spots.roi.value = (0.5, 0.2, 0.5, 0.2)
+        time.sleep(2.0)
+
+        # move spot
+        helper.image.unsubscribe(self._on_image)
+        helper.is_active.value = False
+        spots.is_active.value = False
+
+        self.assertIsNotNone(self._image, "No data received")
+        self.assertIsInstance(self._image, model.DataArray)
+        self.assertIsNotNone(self._image.metadata['Acquisition date'], "No metadata")
```

Comment:
Instead of explicitly writing `'Acquisition date'`, write `model.MD_ACD_DATE`, it reduces the chances of typos.